src/cli_plugin/install/opensearch_dashboards.js

```diff
@@ -47,6 +47,18 @@ export function existingInstall(settings, logger) {
   } catch (e) {
     if (e.code !== 'ENOENT') throw e;
   }
+  if (settings.plugins[0].folderName !== '') {
```

Comment:
The duplicate check for both `.id` and `.folderName` is telling me that there should only be one check here, instead of `.id` it would be using `.targetFolderName()`.

Reply:
The check would be needed for both `id` and `folderName` to check if the plugin is already installed with either the id or the folderName, since the previous install could be with either of the two. This check is added so that the plugin is not installed twice.